include/vast/Dialect/HighLevel/HighLevelOps.td

```diff
@@ -633,28 +667,50 @@ class ArithBinOp< string mnemonic, list< Trait > traits = [] >
 
         The custom assembly form of the operation is as follows:
 
-        %result = <op> %lhs, %rhs  : type
+        %result = <op> %lhs, %rhs  : functional-type(operands, results)
     }];
 
-    let assemblyFormat = [{ $lhs `,` $rhs attr-dict `:` type($result) }];
+    let skipDefaultBuilders = 1;
+    let builders = [ OpBuilder<(ins "Value":$lhs, "Value":$rhs), [{
+        $_state.addOperands(lhs);
+        $_state.addOperands(rhs);
+        auto type = lhs.getType();
+        if(rhs.getType().isa< hl::PointerType >())
+        {
+            // Both operands are pointers -> it's a subtraction
+            if(type.isa< hl::PointerType >())
+                type = odsBuilder.getType< hl::LongType >();
```

Comment:
this is not correct we need to go through our type builder

I suggest to take result type in constructor and transform it before in visitor